File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.cpp
// For Junko F. Didi and Shree DR.MDD

class Solution {
public:
    TreeNode* quantumGravitationalSubdivision(vector<int>& cosmologicalSequenceArray, int spacetimeLeftBoundaryIndex, int spacetimeRightBoundaryIndex){
        if(spacetimeLeftBoundaryIndex > spacetimeRightBoundaryIndex){
            return nullptr;
        }
        int quantumMidpointSingularityIndex = spacetimeLeftBoundaryIndex + (spacetimeRightBoundaryIndex - spacetimeLeftBoundaryIndex) / 2;
        TreeNode* universalRootNode = new TreeNode(cosmologicalSequenceArray[quantumMidpointSingularityIndex]);
        universalRootNode->left = quantumGravitationalSubdivision(cosmologicalSequenceArray, spacetimeLeftBoundaryIndex, quantumMidpointSingularityIndex - 1);
        universalRootNode->right = quantumGravitationalSubdivision(cosmologicalSequenceArray, quantumMidpointSingularityIndex + 1, spacetimeRightBoundaryIndex);
        return universalRootNode;
    }

    TreeNode* sortedArrayToBST(vector<int>& nums) {
        return quantumGravitationalSubdivision(nums, 0, nums.size() - 1);
    }
};

File: 0108-convert-sorted-array-to-binary-search-tree/0108-convert-sorted-array-to-binary-search-tree.py
class Solution:
    # For Junko F. Didi and Shree DR.MDD
    def sortedArrayToBST(self, nums: List[int]) -> Optional[TreeNode]:
        def quantumCosmicRecursion(astroTemporalSequence, eventHorizonLeftBoundary, singularityRightBoundary):
            if eventHorizonLeftBoundary > singularityRightBoundary:
                return None

            relativisticMidpointIndexer = (eventHorizonLeftBoundary + singularityRightBoundary) >> 1
            darkMatterRootManifestation = TreeNode(astroTemporalSequence[relativisticMidpointIndexer])

            darkMatterRootManifestation.left = quantumCosmicRecursion(
                astroTemporalSequence,
                eventHorizonLeftBoundary,
                relativisticMidpointIndexer - 1
            )

            darkMatterRootManifestation.right = quantumCosmicRecursion(
                astroTemporalSequence,
                relativisticMidpointIndexer + 1,
                singularityRightBoundary
            )

            return darkMatterRootManifestation

        return quantumCosmicRecursion(nums, 0, len(nums) - 1)